### PR TITLE
add editorconfig, closes #1129

### LINKED ISSUE #1129: Introduction of editorconfig to assist in code styling
**Is your feature request related to a problem? Please describe.**
I usually indent my java projects with tabs, so I often have to fight my code editor when working on Thorntail in order to indent with spaces.

**Describe the solution you'd like**
I would like to introduce a `.editorconfig` file to the repository (See the site [here](https://editorconfig.org/)) in order to assist with code styling as it is supported in almost all major IDEs.

I personally have experience with editorconfig and have been using it locally to help me properly format files for PRs on this project already and it works very well in the editors I use most:
- NetBeans (via [this](https://github.com/welovecoding/editorconfig-netbeans) plugin)
- VS Code (via [this](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) plugin)

For a complete list of supported editors/IDEs, see their page on it [here](https://editorconfig.org/#download).

** Starter Template ** (based on what I've seen in the project)
```
root = true

[*]
insert_final_newline = true
trim_trailing_whitespace = true
charset = utf-8

[*.java]
indent_style = space
indent_size = 4
curly_bracket_next_line = false
spaces_around_operators = true
continuation_indent_size = 8
indent_brace_style = Java

[*.xml]
indent_style = space
indent_size = 2
indent_brace_style = K&R
```

### PR DESCRIPTION
**Pull request by kg6zvp**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [ ] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [x] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+root = true
+
+[*]
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
```
Comment (Ladicek) [inline .editorconfig:6]: Do we want to add `end_of_line = lf`? I'd say yes.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+[*.java]
+indent_style = space
+indent_size = 4
+curly_bracket_next_line = false
+spaces_around_operators = true
+continuation_indent_size = 8
+indent_brace_style = Java
```
Comment (Ladicek) [inline .editorconfig:15]: Are the last 4 items IntelliJ-specific? I could only find them in https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#ideas-for-domain-specific-properties

Comment (kg6zvp) [reply .editorconfig:15]: They are, but they don't interfere with other editors. I don't usually use IntelliJ and they don't cause a problem. However, it's always good to be as specific as possible is my thought.

File: .editorconfig (hunk 1 above)
Comment (kg6zvp) [reply .editorconfig:6]: Agreed. Added.

**Comment by kg6zvp** [comment]
@Ladicek okay to merge?

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**